aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -38,10 +41,25 @@ class CAFFE2_API Dispatcher final {
 private:
   struct OperatorDef final {
     explicit OperatorDef(FunctionSchema&& schema)
-    : op(std::move(schema)), refcount(0) {}
+    : op(std::move(schema)) {}
+
+    explicit OperatorDef(OperatorName&& op_name)
+    : op(std::move(op_name)) {}
 
     impl::OperatorEntry op;
-    size_t refcount;
+
+    // These refer to the number of outstanding RegistrationHandleRAII
+    // for this operator.  def_count reflects only def() registrations
+    // (in the new world, this should only ever be 1, but old style
+    // registrations may register the schema multiple times, which
+    // will increase this count).  def_and_impl_count reflects the number
+    // of combined def() and impl() registrations.  When a def() gets
```

Comment:
when **the last** def() get's unregistered, we must call the deregistration listeners. Not for the ones before that.